state/src/impls/top_level.rs

```diff
@@ -314,6 +316,7 @@ impl TopLevelState {
         parent_block_number: BlockNumber,
         parent_block_timestamp: u64,
         current_block_timestamp: u64,
+        transaction_location: Option<TransactionLocation>,
```

Comment:
Why is the location optional?

Reply:
Okay. I see the e2e tests.
Some e2e tests postulate the scenario that transactions in the same block. So it can make testing what you really want to do impossible to make all transaction index in the e2e test zero.
But, fixing all e2e tests can be a time-consuming job. How about leaving issue and TODO comment here and making another PR for it?

Reply:
I assumed that chain_executeTransaction RPC will use the function with `None`. If we can ensure that all transactions have an index, I don't want to use the Option.

Similar:
https://github.com/CodeChain-io/foundry/pull/388/files#r426344706

Reply:
How about adding the index as an argument of the RPC or fixing the transaction index as a constant value(I think 0 is reasonable) if the transaction is executed by the RPC.

Reply:
Or how about making the argument of the RPC optional and converting it to zero if the argument is null?

Reply:
Passing a index or using 0 as a default value looks good to me.
I'll change this PR like this: https://github.com/CodeChain-io/foundry/pull/388#discussion_r426347279